samples/drivers/virtualization/ivshmem/doorbell/README.rst

```diff
@@ -0,0 +1,124 @@
+IVSHMEM Doorbell Sample Application
+###################################
+
+Overview
+********
+
+This sample shows how two processes on different operating systems can
+communicate using ivshmem. This is a subset of the functionality provided by
+OpenAMP.
+
+Prerequisites
+*************
+
+QEMU needs to available.
+
+ivshmem-server needs to be available and running. The server is available in
+Zephyr SDK or pre-built in some distributions. Otherwise, it is available in
+QEMU source tree.
+
+ivshmem-client needs to be available as it is employed in this sample as an
+external application. The same conditions of ivshmem-server applies to the
+ivshmem-server, as it is also available via QEMU.
+
+Building and Running
+********************
+
+Building ivshmem-doorbell is as follows:
+
+qemu_cortex_a53
+===============
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/drivers/virtualization/ivshmem/doorbell
+   :host-os: unix
+   :board: qemu_cortex_a53
+   :goals: run
+   :compact:
+
+qemu_kvm_arm64
+==============
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/drivers/virtualization/ivshmem/doorbell
+   :host-os: unix
+   :board: qemu_kvm_arm64
+   :goals: run
+   :compact:
+
+qemu_x86_64
+===========
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/drivers/virtualization/ivshmem/doorbell
+   :host-os: unix
+   :board: qemu_x86_64
+   :goals: run
+   :compact:
+
+How to
+******
+
+.. note::
+
+   The ivshmem shared memory can be manipulated to crash QEMU and bring down
+   Zephyr. Check security section for details.
+
+.. note::
+
+   Due to limited RAM memory available in qemu_x86_64 dts, it is not possible
+   to use the default shared memory size of ivshmem (4MB) for qemu_x86_64.
+
+Steps to reproduce this sample:
+
+- Run ivshmem-server. If vectors != 2, you need to change ivshmem driver
+  :kconfig:option:`CONFIG_IVSHMEM_MSI_X_VECTORS`. Depending on the shared
+  memory size chosen, you may need to adapt the project configuration.
+- Appropriately set ownership of ``/dev/shm/ivshmem`` and
+  ``/tmp/ivshmem_socket`` for your testing scenario.
+- Run Zephyr.
+- Write a message in the shared memory, for instance:
+
+.. code-block:: console
+
+    ./write_shared_memory.sh -m "your message" -s <size of shared memory>
+
+- The shared memory size *must* be kept the same as specified for
+  ivshmem-server. This is the purpose of the ``write_shared_memory`` script;
+  Failing to respect the shared memory size may lead to a QEMU crash.
+- Send an interrupt to the guest. Using ivshmem-client, for instance:
+
+.. code-block:: console
+
+    # find out client id; for this sample, let us assume client id is 0
+    dump
+    # sending a interrupt to client 0 vector 0
+    int 0 0
+
+- The sample will print the text in the shared memory whenever an interrupt is
```

Comment:
You can add code block here to show the informations that sample print. For example：
```
$ sudo west build -t run
-- west build: running target run
[0/1] To exit from QEMU enter: 'CTRL+a, x'[QEMU] CPU: cortex-a53
*** Booting Zephyr OS build zephyr-v3.3.0-1726-g9e7aa68b407c ***
Use write_shared_memory.sh and ivshmem-client to send a message
received IRQ and full message: This is drivers/virtualization/ivshmem/doorbell sample
```